File: cs12b/pa5/backup/DictionaryTest.c
//Shayan Salsabilian
//ssalsabi
//3/12/19
//12B
//Tests for Dictionary.c
//DictionaryTest.c
#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include<assert.h>
#include"Dictionary.h"

int main(){
	Dictionary A=newDictionary();//create a new Dictionary
	printf("Beginning isEmpty Tests:\n");
	int check=isEmpty(A);//check if the Dictionary is empty when it should be
	printf("	Does isEmpty() handle an empty list correctly?\n");
	if(check==1){//if it is print Passed otherwise print Failed
		printf("		Passed\n");
	}else{
		printf("		Failed\n");
	}
	insert(A, "1", "A");//insert into the Dictionary 
	check=isEmpty(A);//then check if the Dictionary is not empty which it should be
	printf("	Does isEmpty() handle a non empty list correctly?\n");
	if(check==0){//if it is print Passed otherwise print failed
		printf("		Passed\n");
	}else{
		printf("		Failed\n");
	}
	printf("Beginning size Tests:\n");
	printf("	Does size() handle correctly when there is one item in the list?\n");
	check=size(A);//get the size, should be 1
	if(check==1){//if it is print passed otherwise print failed
		printf("		Passed\n");
	}else{
		printf("		Failed\n");
	} 
	insert(A, "23", "6");//insert two more items into the list
	insert(A, "T", "ME");
	check=size(A);//check the size again should be 3
	printf("	Does size() handle correctly when there are multiple items in the list?\n");
	if(check==3){//if it is print passed otherwise print failed
		printf("		Passed\n");
	}else{
		printf("		Failed\n");
	}
	printf("Beginning lookup Tests:\n");
	char* check1=lookup(A, "23");//use lookup to find the value needed from the key, should be 6
	printf("	Does lookup() return the right value in the middle of the list?\n");
	if((strcmp(check1, "6"))==0){//if it is print Passed otherwise print Failed
		printf("		Passed\n");
	}else{
		printf("		Failed\n");
	}
	check1=lookup(A, "T"); //use lookup to find the value needed from the key, should be ME
	printf("	Does lookup() return the right value at the end of the list?\n");
	if((strcmp(check1, "ME"))==0){//if it is print Passed otherwise print Failed
		printf("		Passed\n");
	}else{
		printf("		Failed\n");
	}
	printf("Beginning insert Tests:\n");
	printf("	Does insert() work correctly at the end of the list?\n");
	check1=lookup(A, "1");//lookup the value for the key in the Dictionary
	if((strcmp(check1, "A"))==0){//if it matches, then we inserted properly
		printf("		Passed\n");
	}else{
		printf("		Failed\n");
	}
	insert(A, "ABC", "123");//insert an item at the front of the list
	check1=lookup(A, "ABC");//make sure it was properly inserted
	printf("	Does insert() work correctly at the front of the list?\n");
	if((strcmp(check1, "123"))==0){//if it matches, then we inserted properly
		printf("		Passed\n");
	}else{
		printf("		Failed\n");
	}
	/*printf("	The following test should produce an error:\n");
	  insert(A, "ABC", "NOTIMPORTANT");*/
	printf("Beginning delete Tests:\n");
	printf("	Does delete() delete properly from the front of the list?\n");
	delete(A, "ABC");//delete the first item in the list
	check1=lookup(A, "ABC");//lookup the value of the key, it should be null
	if(check1==NULL){//if it is print Passed, otherwise print Failed
		printf("		Passed\n");
	}else{
		printf("		Failed\n");
	}
	printf("	Does delete() properly from the back of the list?\n");
	delete(A, "1");	//delete the last item in the list
	check1=lookup(A, "1");//lookup the value of the key, it should be null
	if(check1==NULL){//if it is print Passed, otherwise print Failed
		printf("		Passed\n");
	}else{
		printf("		Failed\n");
	}
	/*printf("	The following test should produce an error\n");
	  delete(A, "NOTIMPORTANT");*/
	printf("Beginning makeEmpty Tests:\n");
	makeEmpty(A);//empty the list
	printf("	Does makeEmpty make the number of items in the list 0 correctly?\n");
	if((size(A))==0){//check to see if the number of items is now 0, if it is print passed otherwise print failed
	printf("		Passed\n"); 
	}else{
	printf("		Failed\n");
	}
	printf("Beginning print Tests:\n");
	insert(A, "1", "12");//insert three items into the list
	insert(A, "key", "value");
	insert(A, "3", "28");
	printf("The following test should print:\n");
	printf("key value\n");
	printf("3 28\n");
	printf("1 12\n");
	printf("output:\n");
	printDictionary(stdout, A);//print out the list
	return 0;
}

File: cs12b/pa3/DictionaryTest.java
//Shayan Salsabilian
//ssalsabi
//2/13/19
//12B
//Tests the basic functions of our Dictionary Linked Lists
//DictionaryTest.java
public class DictionaryTest
{
  public static void main (String[]args)
  {
    String x;
    Dictionary B = new Dictionary ();	//declaring a Dictionary node
      System.out.println ("Beginning isEmpty Tests:");
    Boolean Check = B.isEmpty ();	//checking while isEmpty() on a empty Dictionary
      System.out.
      println
      ("	Test 1: Does isEmpty handle a Empty Dictionary correctly?");
    if (Check == true)
      {
	System.out.println ("		Passed");
      }
    else
      {
	System.out.println ("		Failed");
      }
    System.out.
      println
      ("	Test 2: Does isEmpty() handle a Dictionary with an item correctly?");
    B.insert ("1", "A");	//inserting an item in the list
    Boolean Check2 = B.isEmpty ();	//Checking that isEmpty() returns false when we have an item
    if (Check2 == false)
      {
	System.out.println ("		Passed");
      }
    else
      {
	System.out.println ("		Failed");
      }
    System.out.println ("Beginning Size Tests:");
    B.delete ("1");		//deleting an item from the list
    System.out.
      println
      ("	Test 1: Does size handle an empty dictionary correctly?");
    if (B.size () == 0)
      {				//checking it returns correctly when our Dictionary is size 0
	System.out.println ("		Passed");
      }
    else
      {
	System.out.println ("		Failed");
      }
    B.insert ("1", "ABC");	//inserting two items into the list
    B.insert ("82", "IAMIRONMAN");
    System.out.
      println
      ("	Test 2: Does size handle a dictionary of size 2 correctly?");
    if (B.size () == 2)
      {				//checking the dictionary is size 2
	System.out.println ("		Passed");
      }
    else
      {
	System.out.println ("		Failed");
      }
    System.out.println ("Beginning Lookup Tests:");
    B.insert ("5", "what did the fox Say");	//inserting another item in the last
    System.out.
      println
      ("	Test 1: Does lookup get the right value from the end of the dictionary?");
    x = B.lookup ("5");		//looking up key 5
    if (x == "what did the fox Say")
      {				//checking if the value matches
	System.out.println ("		Passed");
      }
    else
      {
	System.out.println ("		Failed");
      }
    System.out.
      println
      ("	Test 2: Does lookup get the right value from the beginning of the dictionary?");
    x = B.lookup ("1");		//looking up key 1
    if (x == "ABC")
      {				//checking if the value comes back correctly
	System.out.println ("		Passed");
      }
    else
      {
	System.out.println ("		Failed");
      }
    System.out.
      println
      ("	Test 3: Does lookup return null when the key doesnt exist?");
    x = B.lookup ("AAAAAAAAAAAAAAAAAAAAAAAAAA");	//looking up a nonexistent key
    if (x == null)
      {				//checking if the value comes back null
	System.out.println ("		Passed");
      }
    else
      {
	System.out.println ("		Failed");
      }
    System.out.println ("Beginning Insert Tests:");
    B.insert ("7", "123");	//inserting a key and value into the list
    x = B.lookup ("7");		//looking up to see the value at the key is correct
    System.out.
      println ("	Test 1: Does insert properly insert onto the list?");
    if (x == "123")
      {				//checking it to see that it matches the value it needs to be
	System.out.println ("		Passed");
      }
    else
      {
	System.out.println ("		Failed");
      }
    System.out.
      println
      ("	Test 2: Does insert properly insert to the front of the dictionary?");
    x = B.lookup ("1");		//looking up the first value we entered
    if (x == "ABC")
      {				//checking to see if it matches the value that it should be
	System.out.println ("		Passed");
      }
    else
      {
	System.out.println ("		Failed");
      }
    /*System.out.println("  Test 3: Testing DuplicateKeyException, this should produce a DuplicateKeyException error message and crash the program");
       B.insert("7", "123"); //we already inserted this key so should produce an error
     */
    System.out.println ("Beginning Delete Tests:");
    System.out.
      println
      ("	Test 1: Does delete properly remove from the front of the list?");
    B.delete ("1");		//deleting the first item in the list
    x = B.lookup ("1");		//checks to make sure that key 1 is now holding value null
    if (x == null)
      {				//it should be null, if its not then deletes not working properly
	System.out.println ("		Passed");
      }
    else
      {
	System.out.println ("		Failed");
      }
    B.delete ("7");		//delete the last item in the list
    x = B.lookup ("7");		//the key should no longer have the value
    System.out.
      println
      ("	Test 2: does delete properly remove from the end of the list?");
    if (x == null)
      {				//it should be null, if its not then deletes not working properly
	System.out.println ("		Passed");
      }
    else
      {
	System.out.println ("		Failed");
      }
    /*System.out.println("  Test 3: Testing KeyNotFoundException, this should produce error message involving KeyNotFoundException and crash the program");
       B.delete("8888"); //this key doesn't exist, so the KeyNotFoundException error should be produced
     */
    System.out.println ("Beginning makeEmpty Tests:");
    B.makeEmpty ();		//empties the dictionary
    System.out.
      println ("	Test 1: Does makeEmpty make the dictionary empty?");
    if (B.size () == 0)
      {				//checks to make sure the size of the dictionary is 0
	System.out.println ("		Passed");
      }
    else
      {
	System.out.println ("		Failed");
      }
    System.out.println ("Beginning toString Tests:");
    System.out.println ("The following test should match this:");
    System.out.println ("1 82");	//the values that we will be inserting into the dictionary
    System.out.println ("$ 76");
    System.out.println ("T 3.14");
    System.out.println ("Output:");
    B.insert ("1", "82");	//inserting into the dictionary
    B.insert ("$", "76");
    B.insert ("T", "3.14");
    System.out.println (B);	//printing out the entire dictionary

  }
}
